canonical-web-and-design/maas-ui
# Display a menu button on release notifications

## Done

- Update notifications to be able to display menu buttons on release notifications.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)

### QA steps

- Create a release notification:
```shell
maas $PROFILE notifications create ident='release_notification' category='info' users=true admins=true message='<strong>We’ve released MAAS 2.8.</strong> This version supports LXD VM hosts, faster UI and many bug fixes. Find out <a href="https://maas.io/docs/release-notes">what’s new in 2.8</a>.'
```
- If you dismiss the notification and want to be able to see it again you can run:
```shell
make harness
from maasserver.models.notification import Notification, NotificationDismissal
NotificationDismissal.objects.filter(notification=Notification.objects.get(ident='release_notification')).delete()
```
- Load a React page, you should see the release notification and it should have a chevron next to the dismiss x.

## Fixes

Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/2049.
